iceoryx_posh/test/mocks/base_port_mock.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_POSH_MOCKS_BASE_PORT_MOCK_HPP
+#define IOX_POSH_MOCKS_BASE_PORT_MOCK_HPP
+
+#include "iceoryx_posh/internal/popo/ports/base_port.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using ::testing::_;
+
+class MockBasePort
+{
+  public:
+    using MemberType_t = iox::popo::BasePortData;
+    MockBasePort(MemberType_t*){};
+    MockBasePort() = default;
+
+    MockBasePort(const MockBasePort&) = delete;
+    MockBasePort& operator=(const MockBasePort&) = delete;
+    MockBasePort(MockBasePort&&) noexcept
+    {
+    }
+    MockBasePort& operator=(MockBasePort&&) noexcept
+    {
+        return *this;
+    }
```

Comment:
clang tidy warning here since you technical violate the rule of 5 when there is not dtor. Could you therefore please add `~MockBasePort() =default`